### DISCUSSION TITLE
Multiple farmers, same wallet

### DISCUSSION THREAD

**Discussion by iWedmak** [Chia-Network/chia-blockchain#4034] [Q&A]
I need to start two farmers on different servers in different data-centers, without direct connection between them.
i want one farmer to have all my keys (public and private) and i want second one to have only public keys.
is it gonna be sufficient to run ` chia init -c [directory] ` on second one for him to be able to farm for my wallet without having full access to wallet?

**Comment by SackofFluor** [comment]
I would just run both as a full-node with the same key phrases.
To be honest this seemed to be the easiest way to connect multiple mining locations to one wallet.

**Reply by iWedmak** [reply]
sure, but not really secure enough, i do not trust second location to have private keys

**Reply by aslangencer** [reply]
If both machines are equally secure and they belong to same person/party, why not? In fact it even serves as storing a backup of the private key. no?

**Comment by happycouak** [comment]
I have the same needs, actually I have tried initializing chia without starting any process, and -f "public_farmer_key" and -p "pool_public_keys", but I have this error when I check retreived plot on my main node:

```
2021-05-07T08:17:00.803  chia.plotting.plot_tools         : ERROR    Failed to open file /nfs2/plot-k32-2021-05-07-00-48-ead8024ae6d716eac7e7d3f10f66680ff9186c88cb7c97eabfeaa375be4c23ff.plot. Invalid number of bytes 2 Traceback (most recent call last):
  File "/home/ubuntu/chia-blockchain/chia/plotting/plot_tools.py", line 202, in process_file
    ) = parse_plot_info(prover.get_memo())
  File "/home/ubuntu/chia-blockchain/chia/plotting/plot_tools.py", line 81, in parse_plot_info
    raise ValueError(f"Invalid number of bytes {len(memo)}")
ValueError: Invalid number of bytes 2
```

So interrested by a solution regarding farming in hostile context (eg without my private keys)

**Comment by happycouak** [comment]
Maybe I need to also add the "-a" option with the fingerprint of the main node ?

**Reply by iWedmak** [reply]
dont know, im my case just using public keys with plotting command works

**Reply by happycouak** [reply]
Do you mean that on the remote farmer without full node started, you only use the -p "public_key", and when created plots are sended to the full node they are well handled by "chia plots check" ?

**Reply by iWedmak** [reply]
with this
-f [farmer pk]: This is your "Farmer Public Key". Utilise this when you want to create plots on other machines for which you do not want to give full chia account access. To find your Chia Farmer Public Key use the following command: chia keys show

-p [pool pk]: This is your "Pool Public Key". Utilise this when you want to create plots on other machines for which you do not want to give full chia account access. To find your Chia Pool Public Key use the following command: chia keys show

but yes

**Reply by happycouak** [reply]
Then it doesn't work for me actually

**Reply by iWedmak** [reply]
shame ( try write in issues?

**Reply by happycouak** [reply]
For the sake of completeness, here it is: https://github.com/Chia-Network/chia-blockchain/issues/4212
